hw/ip/aes/pre_sca/alma_post_syn/README.md

```diff
@@ -108,15 +110,91 @@ The other variables usually do not need editing.
    ```
 1. Activate the python environment:
    ```sh
-   source ${COCO_ALMA_TOP}/dev/bin/activate
+   source ${ALMA_TOP}/dev/bin/activate
    ```
 1. Run the verification flow:
    ```sh
    ${REPO_TOP}/hw/ip/aes/pre_sca/alma_post_syn/verify_aes_post_syn.sh
    ```
-   which should print something like below:
+   which should print something like below indicating that verification succeeded:
    ```sh
    ...
    Finished in 0.42
    The execution is secure
    ```
+
+## Incrementally updating the flow
+
+### Format of the simulation model (verilog) file
+The python script `simplify_techlib.py` makes few assumptions on the simulation model file from the techlib:
+
+1. The cell definitions are wrapped with surrounding lines `` `celldefine`` and `` `endcelldefine``.
+These markers are used to split cells during parsing.
+
+1. The cell names have a particular form (e.g. `ABC_NOT2_1`) that appears after `module` where the part after the last underscore refers to the size of the cell.
+The cells `ABC_NOT2_1` and `ABC_NOT2_2` are internally represented with `ABC_NOT2_` by dropping the size parameter, so that the patching effort is not repeated for the cells with the same functional description.
+What is really important about the format is that the script uses a matching regexp to capture it.
+If another techlib uses another format, then the regexp part of the script needs to be updated to match the corresponding format.
+
+An example artificial cell would look like below:
+
+```
+`celldefine
+module ABC_NOT2_1 (Out, In1, In2);
+   output Out;
+   input In1, In2;
+   nand (Out, In1, In2);
+
+   specify
+   // there is usually timing information here
+   // but the flow strips this part out
+   endspecify
+
+endmodule
+`endcelldefine
+```
+
+### Adding complex cell definitions to the patch file
+
+As outline above, the patch file does not provide definitions for all cells, but only those cells from the simulation model that are too complex for Yosys to digest.
+A single syntehsis of AES S-box contains only a small selection of cells from the technology library, and therefore the patch file only addresses the complex cells from this very subset.
```

Comment:
Typo: syntehis -> synthesis
Or better: "The synthesized netlist of a single AES S-Box..."

Reply:
Fixed the typo, but I did not change the ordering here. Putting "a single AES S-box" sounds like patch file changes would only be needed only when we are changing RTL implementation of the AES S-box. At least that's how I would interpret.

However, the emphasis is on the fact that even though the AES S-box remains same (before synthesis), the different synthesis runs are the cause that lead to different selection of cells. Therefore, one probably needs to add more cell definitions into the patch file for each different synthesis, but this effort should become rare when patch file grows.

Reply:
This makes totally sense, thanks for explaining :-)